src/util/async/Operation.hpp

```diff
@@ -181,4 +186,58 @@ using Operation = impl::BasicOperation<Outcome<RetType>>;
 template <typename CtxType, typename OpType>
 using ScheduledOperation = impl::BasicScheduledOperation<CtxType, OpType>;
 
+/**
+ * @brief The `future` side of async operations that automatically repeat until aborted
+ *
+ * @note The current implementation requires the user provided function to return void and to take no arguments. There
+ * is also no mechanism to request the repeating task to stop from inside of the user provided block of code.
+ *
+ * @tparam CtxType The type of the execution context
+ */
+template <typename CtxType>
+class RepeatingOperation {
+    using DataType = std::expected<void, ExecutionError>;
+    using StopSourceType = typename CtxType::StopSource;
+
+    std::unique_ptr<util::Repeat> repeat_;
+    StoppableOutcome<DataType, StopSourceType> outcome_;
+    StoppableOperation<DataType, StopSourceType> op_;
+
+public:
+    /**
+     * @brief Construct a new Repeating Operation object
+     * @note The first invocation of the user-provided function happens with no delay
+     *
+     * @param executor The executor to operate on
+     * @param interval Time to wait before repeating the user-provided block of code
+     * @param fn The function to execute repeatedly
+     */
+    RepeatingOperation(auto& executor, std::chrono::steady_clock::duration interval, std::invocable auto&& fn)
+        : repeat_(std::make_unique<util::Repeat>(executor)), op_(&outcome_)
+    {
+        repeat_->start(interval, std::forward<decltype(fn)>(fn));
+    }
+
+    /**
+     * @brief Aborts the operation and the repeating timer
+     * @warning Calling this from inside of the repeating operation yields a deadlock
+     */
+    void
+    abort() noexcept
+    {
+        op_.requestStop();
+        repeat_->stop();
+    }
+
+    /**
+     * @brief Wait for the operation to get cancelled
+     * @warning Calling this from inside of the repeating operation yields a deadlock
```

Comment:
This can be worked around with custom code instead of `util::Repeat` but i decided not to make it more complicated than is necessary now. We can always improve it in the future as long as the interface remains the same or does not change too much.